nixos/modules/system/etc/setup-etc.pl

```diff
@@ -108,6 +108,8 @@ sub link {
             my $uid = read_file("$_.uid"); chomp $uid;
             my $gid = read_file("$_.gid"); chomp $gid;
             copy "$static/$fn", "$target.tmp" or warn;
+            $uid = getpwnam $uid unless $uid =~ /^\d+$/;
+            $gid = getgrnam $gid unless $gid =~ /^\d+$/;
```

Comment:
mixing username and uid can have tricky [edge cases](https://github.com/systemd/systemd/issues/6237#issue-239378060). It is probably better to make to options for group/user and make them mutual exclusive with uid/gid.